### PR TITLE
pycaw #15: Ignore properties that cause a COMError

### PR DESCRIPTION
**Pull request by reversefold**
Fixes issues #15 by catching the specific exception and adding a warning. Test also added which passes for me.

### CODE REVIEW

**Comment by reversefold** [comment]
Not sure why the TravisCI build failed, but it doesn't seem related to my changes.

File: tests/test_core.py [hunk 1]
```diff
@@ -46,6 +48,22 @@ def test_device_unicode(self):
...
+        """
+
+        with captured_output() as (out, err):
+            dev = mock.Mock()
+            dev.GetId = mock.Mock(return_value="id")
+            dev.GetState = mock.Mock(return_value=AudioDeviceState.Active)
+            store = mock.Mock()
+            store.GetCount = mock.Mock(return_value=3)
+            store.GetAt = mock.Mock(return_value="pk")
+            store.GetValue = mock.Mock(side_effect=_ctypes.COMError(None, None, None))
```
Comment (AndreMiras) [inline tests/test_core.py:63]: Linter is complaining about that line:
```
.\tests\test_core.py:63:80: E501 line too long (86 > 79 characters)
```

File: tests/test_core.py [hunk 2]
```diff
@@ -46,6 +48,22 @@ def test_device_unicode(self):
...
+            dev.OpenPropertyStore = mock.Mock(return_value=store)
+            audev = AudioUtilities.CreateDevice(dev)
```
Comment (AndreMiras) [inline tests/test_core.py:65]:
```
.\tests\test_core.py:65:13: F841 local variable 'audev' is assigned to but never used
```

File: tests/test_core.py [hunk 3]
```diff
@@ -46,6 +48,22 @@ def test_device_unicode(self):
             for device in devices:
                 print("device: %s" % device)
 
+    def test_device_failed_properties(self):
+        """
+        Makes sure printing a device doesn't crash.
+        """
+
+        with captured_output() as (out, err):
```
Comment (AndreMiras) [inline tests/test_core.py:56]: could we potentially assert on what got `out`?

File: tests/test_core.py (hunk 2 above)
Comment (AndreMiras) [inline tests/test_core.py:65]: I think only this line needs to be in the `captured_output` context manager, correct?

File: tests/test_core.py [hunk 4]
```diff
@@ -46,6 +48,22 @@ def test_device_unicode(self):
             for device in devices:
                 print("device: %s" % device)
 
+    def test_device_failed_properties(self):
+        """
+        Makes sure printing a device doesn't crash.
+        """
```
Comment (AndreMiras) [inline tests/test_core.py:54]: This could potentially be a single line
https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

**Review by AndreMiras — COMMENTED** [review]
LGTM, I left a few comment on the unit tests part, but implementation looks good to me, thanks

**Comment by reversefold** [comment]
Updated

**Review by AndreMiras — APPROVED** [review]
Very nice, thanks for addressing the comments

**Pull request merged**